### PR TITLE
Feature/List Detection Employees w/ Filtering

### PR DESCRIPTION
**Pull request by unparalleled-js**
`list` commands for `departing-employee` and `high-risk-employee`, really useful for checking the occupants of these lists w/o having to open the UI.

Test:

`code42 departing-employee list`
`code42 departing-employee list -f RAW-JSON`
`code42 departing-employee list -f CSV`

`code42 high-risk-employee list`
`code42 high-risk-employee list -f RAW-JSON`
`code42 high-risk--employee list -f CSV`

Questions:

Do we want to echo something like "nobody on list" when 0 results?

### CODE REVIEW

**Comment by alanag13** [comment]
> Do we want to echo something like "nobody on list" when 0 results?

Probably, since this is an established pattern in other places where a list yields no results. I would say something like `There are currently no users on the <listname> list.`

File: src/code42cli/cmds/departing_employee.py [hunk 1]
```diff
@@ -25,6 +27,18 @@ def departing_employee(state):
     pass
 
 
+@departing_employee.command("list")
+@format_option
+@sdk_options()
+def _list(state, format):
+    employee_generator = _get_departing_employees(state.sdk)
+    employee_list = [
+        employee for employees in employee_generator for employee in employees["items"]
```
Comment (alanag13) [inline src/code42cli/cmds/departing_employee.py:36]: I would prefer if we avoid putting multiple instances of `for` on the same line.

File: src/code42cli/cmds/departing_employee.py [hunk 2]
```diff
@@ -25,6 +27,18 @@ def departing_employee(state):
...
+    ]
+    formatter = OutputFormatter(format, HEADER_KEYS_MAP)
```
Comment (alanag13) [inline src/code42cli/cmds/departing_employee.py:38]: should departing employees also include the departure date in the output?

**Review by kiran-chaudhary — CHANGES_REQUESTED** [review]
Newline character in table format makes the output look weird a bit.
```
Username                             Notes                          Departure Date   
juliya.smith+partners@code42.com     test                           None             
spatel@code42.com                    Test workato test test         2020-10-20       
test-1603479510@test.com             overwrite notes
update notes   2020-11-02       
test-1603479374@test.com             overwrite notes
update notes   2020-11-02       
test-1603479222@test.com             overwrite notes
update notes   2020-11-02       
test-1603478993@test.com             overwrite notes

```

**Comment by kiran-chaudhary** [comment]
An option to filter results by departing date too can be useful. Use case: A user is interested in knowing employees departing in next 1 month or between given dates.

**Comment by unparalleled-js** [comment]
@kiran-chaudhary Thank you for your feedback! You found some good stuff, I addressed it all I believe except I made a compromise.  Instead of implementing some custom filtering via Departure Date, I added the API-given filter clauses as options.  They include `EXFILTRATION_24_HOURS` and `EXFILTRATION_30_DAYS` and departing employee even has one for `LEAVING_TODAY`... So these mirror the same functionality in the UI!  Thanks Kiran, let me know what you think

**Comment by kiran-chaudhary** [comment]
> @kiran-chaudhary Thank you for your feedback! You found some good stuff, I addressed it all I believe except I made a compromise. Instead of implementing some custom filtering via Departure Date, I added the API-given filter clauses as options. They include `EXFILTRATION_24_HOURS` and `EXFILTRATION_30_DAYS` and departing employee even has one for `LEAVING_TODAY`... So these mirror the same functionality in the UI! Thanks Kiran, let me know what you think

This looks good now. 

Below points are not on our team's plate but a general suggestion to be taken up with concerned team:

- Analysing the data it seems there should be an option to filter out results of past dates i.e right now "departing employee" displays data even of "departed employees", if not added then `OPEN` filter will display current data even in 2022.

- As we renamed our product from "Insider threat" to "Insider Risk", on that same thought process, we should recommend renaming "EXFILTRATION_24_DAYS" to a milder word like "DEPARTING_WITHIN_24_DAYS".

File: src/code42cli/cmds/departing_employee.py [hunk 3]
```diff
@@ -12,10 +14,17 @@
...
 from code42cli.file_readers import read_csv_arg
 from code42cli.file_readers import read_flat_file_arg
+from code42cli.options import format_option
 from code42cli.options import sdk_options
 
 
 DATE_FORMAT = "%Y-%m-%d"
+filter_option = click.option(
+    "--filter",
+    help="Departing employee filter options.",
```
Comment (kiran-chaudhary) [inline src/code42cli/cmds/departing_employee.py:24]: Help message should specify the default option.

Comment (kiran-chaudhary) [reply src/code42cli/cmds/departing_employee.py:24]: `ALL` seems more relevant than 'OPEN', right now this is mirroring API's naming convention so its fine, just a thought if we could create a map of custom names against the choices.

**Review by kiran-chaudhary — APPROVED** [review]
LGTM, Just one concern to address the default option in help text.

**Comment by unparalleled-js** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request closed without merging**